File: samples/conference/registration-bc/src/main/java/com/microsoft/conference/registration/readmodel/QueryServices/Order.cs
using System;
using System.Collections.Generic;

namespace Registration.ReadModel
{
    public class Order
    {
        private IList<OrderLine> _lines = new List<OrderLine>();

        public Guid OrderId { get; set; }
        public Guid ConferenceId { get; set; }
        public int Status { get; set; }
        public string RegistrantEmail { get; set; }
        public string AccessCode { get; set; }
        public decimal TotalAmount { get; set; }
        public DateTime? ReservationExpirationDate { get; set; }
        public bool IsFreeOfCharge()
        {
            return TotalAmount == 0;
        }

        public void SetLines(IList<OrderLine> lines)
        {
            _lines = lines;
        }
        public IList<OrderLine> GetLines()
        {
            return _lines;
        }
    }
    public class OrderLine
    {
        public Guid OrderId { get; set; }
        public Guid SeatTypeId { get; set; }
        public string SeatTypeName { get; set; }
        public int Quantity { get; set; }
        public decimal UnitPrice { get; set; }
        public decimal LineTotal { get; set; }
    }
}

File: samples/conference/registration-bc/src/main/java/com/microsoft/conference/registration/readmodel/queryservices/Order.java
package com.microsoft.conference.registration.readmodel.queryservices;

import java.math.BigDecimal;
import java.util.ArrayList;
import java.util.Date;
import java.util.List;

public class Order {
    public String orderId;
    public String conferenceId;
    public int status;
    public String registrantEmail;
    public String accessCode;
    public BigDecimal totalAmount;
    public Date reservationExpirationDate;
    private List<OrderLine> orderLines = new ArrayList<>();

    public boolean IsFreeOfCharge() {
        return totalAmount.equals(BigDecimal.ZERO);
    }

    public void SetLines(List<OrderLine> lines) {
        orderLines = lines;
    }

    public List<OrderLine> GetLines() {
        return orderLines;
    }
}